### PR TITLE
Fully adopt `Sendable`

### PR DESCRIPTION
**Pull request by dnadoba**

### CODE REVIEW

File: Sources/NIOSSH/Connection State Machine/Operations/AcceptsUserAuthMessages.swift [hunk 1]
```diff
@@ -62,9 +62,11 @@ extension AcceptsUserAuthMessages {
...
         if let future = result {
-            var banner: SSHServerConfiguration.UserAuthBanner?
+            let banner: SSHServerConfiguration.UserAuthBanner?
             if case .server(let config) = role {
                 banner = config.banner
+            } else {
+                banner = nil
             }
 
             return .possibleFutureMessage(future.map { Self.transform($0, banner: banner) })
```
Comment (dnadoba) [inline Sources/NIOSSH/Connection State Machine/Operations/AcceptsUserAuthMessages.swift:72]: This fixes a warning with strict concurrency checking enabled

**Review by glbrntt — APPROVED** [review]

**Comment by dnadoba** [comment]
@swift-server-bot test this please

**Review by Lukasa — APPROVED** [review]

**Comment by Lukasa** [comment]
@swift-server-bot test this please

**Comment by dnadoba** [comment]
Allocation counters regressions are unrelated to this PR:
https://github.com/apple/swift-nio-ssh/pull/152

**Comment by dnadoba** [comment]
Allocation have regressed because I marked the static function `transform(:_)` with `@Sendable` to workaround this issue: https://github.com/apple/swift/issues/61449

However, we can also just form a closure manually and it doesn't allocate any more.

**Pull request merged**